src/libraries/System.Net.Http/src/System/Net/Http/SocketsHttpHandler/Http3RequestStream.cs

```diff
@@ -300,13 +304,26 @@ public async Task<HttpResponseMessage> SendAsync(CancellationToken cancellationT
                 _connection.Abort(ex);
                 throw new HttpRequestException(ex.HttpRequestError, SR.net_http_client_execution_error, ex);
             }
+            catch (QPackDecodingException ex)
+            {
+                Exception abortException = _connection.Abort(HttpProtocolException.CreateHttp3ConnectionException(Http3ErrorCode.QPackDecompressionFailed));
+                throw new HttpRequestException(HttpRequestError.InvalidResponse, SR.net_http_invalid_response, ex);
+            }
+            catch (QPackEncodingException ex)
+            {
+                _stream.Abort(QuicAbortDirection.Write, (long)Http3ErrorCode.InternalError);
+                throw new HttpRequestException(HttpRequestError.Unknown, SR.net_http_client_execution_error, ex);
+            }
```

Comment:
How are we deciding between aborting the stream vs connection here?
Or are we expecting `QPackEncodingException` to practically never occur?

Reply:
failure to encode here means we try to send wrong data -> just aborting the (request) stream should be good. This happens in the test which attempts to send non-ascii headers I think.

Failure to decode means server sent corrupted data, which we can't recover from, so we have to tear down entire connection per RFC instructions.